File: otherlang/src/test/java/com/github/tetrisanalyzer/piecegenerator/JavaPieceGeneratorTest.java
package nu.tengstrand.tetrisanalyzer.piecegenerator;

import org.junit.Test;

import static org.junit.Assert.assertTrue;

public class JavaPieceGeneratorTest {

    @Test
    public void nextPiece() {
        PieceGenerator pieceGenerator = new JavaPieceGenerator();

        for (int i=1; i<=1000; i++) {
            int pieceNumber = pieceGenerator.nextPiece().number();

            assertTrue(pieceNumber >= 1);
            assertTrue(pieceNumber <= 7);
        }
    }
}

File: src/test/scala/nu/tengstrand/tetrisanalyzer/piecegenerator/JavaPieceGeneratorTest.scala
package nu.tengstrand.tetrisanalyzer.piecegenerator

import nu.tengstrand.tetrisanalyzer.BaseTest
import org.junit.Test

class JavaPieceGeneratorTest extends BaseTest {
  @Test def nextPiece {
    val pieceGenerator = new JavaPieceGenerator

    for (i <- 1 to 1000) {
      val pieceNumber = pieceGenerator.piece.number.toInt
      pieceNumber should be >= 1
      pieceNumber should be <= 7
    }
  }
}